transport/src/main/java/io/camunda/zeebe/transport/stream/impl/RemoteStreamServiceImpl.java

```diff
@@ -28,14 +28,17 @@ public class RemoteStreamServiceImpl<M extends BufferReader, P extends BufferWri
   private final RemoteStreamerImpl<M, P> streamer;
   private final RemoteStreamEndpoint<M> apiServer;
   private final RemoteStreamRegistry<M> registry;
+  private final RemoteStreamRestarter<P> remoteStreamRestarter;
 
   public RemoteStreamServiceImpl(
       final RemoteStreamerImpl<M, P> streamer,
       final RemoteStreamEndpoint<M> apiServer,
-      final RemoteStreamRegistry<M> registry) {
+      final RemoteStreamRegistry<M> registry,
+      final RemoteStreamRestarter<P> remoteStreamRestarter) {
     this.streamer = streamer;
     this.apiServer = apiServer;
     this.registry = registry;
+    this.remoteStreamRestarter = remoteStreamRestarter;
```

Comment:
:+1: I like the usage of a SRP class to encapsulate sending the request. It might be a little too narrow however. 

:x: I would propose instead that, since the `RemoteStreamEndpoint` already handles communicating over the transport, we rename it `RemoteStreamTransport` or `RemoteStreamCommunication`, and delegate sending the request to it.

This has some additional benefits:

- Centralize network communication of the remote stream
- Since the `RemoteStreamEndpoint` (or now the `RemoteStreamTransport`) is an actor itself, we can have the actor send the stream recreate request. This is good because the `event(ClusterMembershipEvent)` callback is called from the thread of the membership protocol (in this case, the SWIM membership protocol's thread), and we don't want to do more work there to avoid membership update congestion.